lib/ddtrace/contrib/mongodb/subscribers.rb

```diff
@@ -0,0 +1,104 @@
+module Datadog
+  module Contrib
+    # MongoDB module includes classes and functions to instrument MongoDB clients
+    module MongoDB
+      # `MongoCommandSubscriber` listens to all events from the `Monitoring`
+      # system available in the Mongo driver.
+      class MongoCommandSubscriber
+        # rubocop:disable Metrics/MethodLength
+        # rubocop:disable Metrics/AbcSize
+        # rubocop:disable Metrics/CyclomaticComplexity
+        def started(event)
+          pin = Datadog::Pin.get_from(event.address)
+          return unless pin && pin.enabled?
+
+          # start a trace and store it in the current thread; using the `operation_id`
+          # is safe since it's a unique id used to link events together. Also only one
+          # thread is involved in this execution so thread-local storage should be safe. Reference:
+          # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring.rb#L70
+          # https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/monitoring/publishable.rb#L38-L56
+          span = pin.tracer.trace('mongo.cmd', service: pin.service, span_type: Datadog::Ext::Mongo::TYPE)
+          Thread.current[:datadog_mongo_span] = span
+
+          # common fields for all commands
+          command_name = event.command_name
+          collection = event.command[command_name]
+          span.set_tag(Datadog::Ext::Mongo::COLLECTION, collection)
+          span.set_tag(Datadog::Ext::Mongo::DB, event.database_name)
+          span.set_tag(Datadog::Ext::NET::TARGET_HOST, event.address.host)
+          span.set_tag(Datadog::Ext::NET::TARGET_PORT, event.address.port)
+
+          # commands are handled so that specific fields are normalized based on type, if it's
+          # a command that requires documents or a specific query. For some commands, we only
+          # take in consideration the query ('q') to keep the cardinality low
+          # NOTE: 'find' doesn't use a symbol
+          case command_name
+          when 'find'
+            filter = event.command['filter']
+            unless filter.nil? || filter.empty?
+              query = Datadog::Contrib::MongoDB.normalize_query(filter)
+              span.set_tag('mongodb.filter', query)
```

Comment:
not sure if it makes sense having different tags for different commands (`mongodb.filter`, `mongodb.updates` etc...). We can use simply `mongodb.query` adding the command as a tag?

Reply:
let's keep just a `mongodb.query`, probably it's more meaningful for users